mobile/examples/super_resolution/android/app/src/main/java/ai/onnxruntime/example/superresolution/SuperResPerformer.kt

```diff
@@ -27,36 +26,32 @@ internal class SuperResPerformer(
         // Step 2: get the shape of the byte array and make ort tensor
         val shape = longArrayOf(rawImageBytes.size.toLong())
 
-        ortEnv.use {
-            val inputTensor = OnnxTensor.createTensor(
-                ortEnv,
-                ByteBuffer.wrap(rawImageBytes),
-                shape,
-                OnnxJavaType.UINT8
-            )
-            inputTensor.use {
-                // Step 3: call ort inferenceSession run
-                val output = ortSession.run(Collections.singletonMap("image", inputTensor))
-
-                // Step 4: output analysis
-                output.use {
-                    val rawOutput = (output?.get(0)?.value) as ByteArray
-                    val outputImageBitmap =
-                        byteArrayToBitmap(rawOutput)
-
-                    // Step 5: set output result
-                    result.outputBitmap = outputImageBitmap
-                }
+        val inputTensor = OnnxTensor.createTensor(
+            ortEnv,
+            ByteBuffer.wrap(rawImageBytes),
+            shape,
+            OnnxJavaType.UINT8
+        )
+        inputTensor.use {
+            // Step 3: call ort inferenceSession run
+            val output = ortSession.run(Collections.singletonMap("image", inputTensor))
+
+            // Step 4: output analysis
+            output.use {
+                val rawOutput = (output?.get(0)?.value) as ByteArray
+                val outputImageBitmap =
+                    byteArrayToBitmap(rawOutput)
+
+                // Step 5: set output result
+                result.outputBitmap = outputImageBitmap
             }
         }
+        ortSession.close()
```

Comment:
Wouldn't we want the session to stay open for subsequent calls?

Reply:
Yeah - the examples should make it clear you want to create the session once and keep it around as long as you want to do any inferencing. It's an extremely expensive operation, and OrtSession.run is designed to be called multiple times including concurrently.

Reply:
> Do we also need to make these changes for the iOS app?

For the ios app, it's probably fine.  The inference session is only being created once.

Reply:
moved session.close() to `OnDestory(). ` thanks for the catch.